publication-commons/src/test/java/cucumber/permissions/file/FileAccessReadFeatures.java

```diff
@@ -9,40 +12,64 @@
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
+import java.time.Instant;
 import no.unit.nva.model.FileOperation;
 import no.unit.nva.model.associatedartifacts.file.File;
 import no.unit.nva.publication.model.business.UserInstance;
 import no.unit.nva.publication.permissions.file.FilePermissions;
 
 public class FileAccessReadFeatures {
+
+    public static final String EMPTY_PROPERTY = "";
     private final FileScenarioContext scenarioContext;
 
     public FileAccessReadFeatures(FileScenarioContext scenarioContext) {
         this.scenarioContext = scenarioContext;
     }
 
-    @SuppressWarnings("unchecked")
-    @Given("a file in the {string} state")
+    @Given("a file of type {string}")
     public void a_file_in_the_state(String string) throws ClassNotFoundException {
-        var clazz = (Class<File>)Class.forName(File.class.getPackageName() + "." + string);
-        var file = File.builder().build(clazz);
+        var file = getFile(string, EMPTY_PROPERTY);
+        scenarioContext.setPublication(randomNonDegreePublication());
+        scenarioContext.setFile(file);
+    }
+
+    @Given("a file of type {string} with property {string}")
+    public void a_file_in_the_state(String fileType, String fileProperty) throws ClassNotFoundException {
```

Comment:
Maybe use camelCase as we do other places.

Reply:
yeah, I guess this was the autogenerated style. I'll fix

Reply:
also renamed the file to 👍 FileAccessFeatures